cli/service/service.go

```diff
@@ -93,42 +94,69 @@ func Log(serviceName string, tailing bool) {
-		for _, serviceInstance := range runningServices.Instances["active"] {
-			org, name := cutil.SplitOrgSpecUrl(refUrl)
-			if (serviceInstance.SpecRef == name && serviceInstance.Org == org) || strings.Contains(serviceInstance.SpecRef, refUrl) {
-				instanceId = serviceInstance.InstanceId
-				serviceFound = true
-				msgPrinter.Printf("Displaying log messages for service %v with service id %v.", serviceInstance.SpecRef, instanceId)
+		return
+	}
+
+	refUrl := serviceName
+	// Get the list of running services from the agent.
+	runningServices := api.AllServices{}
+
+	cliutils.HorizonGet("service", []int{200}, &runningServices, false)
+	// Search the list of services to find one that matches the input service name. The service's instance Id
+	// is what appears in the syslog, so we need to save that.
+	serviceFound := false
+	var instanceId string
+	org, name := cutil.SplitOrgSpecUrl(refUrl)
+	for _, serviceInstance := range runningServices.Instances["active"] {
+		if (serviceInstance.SpecRef == name && serviceInstance.Org == org) || strings.Contains(serviceInstance.SpecRef, refUrl) {
+			instanceId = serviceInstance.InstanceId
+			serviceFound = true
+			msgPrinter.Printf("Displaying log messages for service %v with service id %v.", serviceInstance.SpecRef, instanceId)
+			msgPrinter.Println()
+			if tailing {
+				msgPrinter.Printf("Use ctrl-C to terminate this command.")
 				msgPrinter.Println()
-				if tailing {
-					msgPrinter.Printf("Use ctrl-C to terminate this command.")
-					msgPrinter.Println()
-				}
-				break
 			}
+			break
 		}
-		if !serviceFound {
-			cliutils.Fatal(cliutils.CLI_INPUT_ERROR, msgPrinter.Sprintf("Service %v is not running on the node.", refUrl))
+	}
+	if !serviceFound {
+		cliutils.Fatal(cliutils.CLI_INPUT_ERROR, msgPrinter.Sprintf("Service %v is not running on the node.", refUrl))
+	}
+
+	// Check service's log-driver to read logs from correct place
+	var nonDefaultLogDriverUsed bool
+	for _, v := range runningServices.Definitions["active"] {
+		def := &persistence.MicroserviceDefinition{}
+		defBytes, _ := json.Marshal(v)
+		if err := json.Unmarshal(defBytes, def); err != nil {
+			cliutils.Fatal(cliutils.CLI_INPUT_ERROR, msgPrinter.Sprintf("Service definition unmarshalling error: %v", err))
 		}
-		if runtime.GOOS == "darwin" {
-			LogMac(instanceId, tailing)
-		} else {
-			LogLinux(instanceId, tailing)
+
+		if (def.SpecRef == name && def.Org == org) || strings.Contains(def.SpecRef, refUrl) {
+			if def.Deployment != "" {
+				deployment := &containermessage.DeploymentDescription{}
+				if err := json.Unmarshal([]byte(def.Deployment), deployment); err != nil {
+					cliutils.Fatal(cliutils.CLI_INPUT_ERROR, msgPrinter.Sprintf("Deployment unmarshalling error: %v", err))
+				}
+
+				for _, service := range deployment.Services {
+					if service.LogDriver != "" && service.LogDriver != "syslog" {
+						cliutils.Verbose(msgPrinter.Sprintf("Non default log-driver used for service: %v", service.LogDriver))
```

Comment:
Change to something like "Service logs are only available for services with 'syslog' log-driver. This service is using log-driver %v".
That way they know which driver supports logs and there's no chance they will think default refers to the docker default not ours.

Reply:
Changed message, thanks